int64 mod
---
int64 log_interval